qml/DepthAxis.qml

```diff
@@ -0,0 +1,128 @@
+import QtQuick 2.7
+import QtQuick.Controls 2.2
+import QtQuick.Layouts 1.3
+
+import Logger 1.0
+
+Item {
+    id: root
+
+    // these are actually all in meters right now
+    property var start_mm: 0
+    property var end_mm: 0
+    property var length_mm: end_mm - start_mm
+    property var depth_mm: 0
+
+    property var validIncrements: [0.1, 0.25, 0.5, 1, 2, 5, 10]
+
+    property int numTicks: 10
+    property int maxTicks: 10
+    property var increment: 1
+
+    property var color: "white"
+
+    function getIncrement() {
+        var inc;
+        for (var i = 0; i < validIncrements.length; i++) {
+            inc = validIncrements[i];
+            if (length_mm / inc <= maxTicks) {
+                break;
+            }
+        }
+        return inc;
+    }
+
+    function recalc() {
+        increment = getIncrement()
+        numTicks = length_mm / increment + 1
+    }
+
+    onStart_mmChanged: {
+        recalc()
+    }
+
+    onEnd_mmChanged: {
+        recalc()
+    }
+
+    function nominalHeight() {
+        return increment * parent.height / length_mm
+    }
+
+    function remainingHeight() {
+        return parent.height - ((numTicks-1) * nominalHeight())
+    }
+
+    Component {
+        id: tickMark
+        RowLayout {
+            anchors.right: parent.right
+            height: lastIndex ? remainingHeight() : nominalHeight()
+            Layout.fillHeight: true
+            Layout.alignment: Qt.AlignRight | Qt.AlignBottom
+
+            property var lastIndex: (index + 1) == numTicks
+
+            Rectangle {
+                id: tick
+                width: 15
+                height: 2
+                Layout.alignment: Qt.AlignRight | Qt.AlignBottom
+                color: root.color
+                border.color: root.color
+                border.width: 1
+                visible: parent.height > 0
+            }
+        }
+    }
+
+    Component {
+        id: tickLabel
+        RowLayout {
+            anchors.right: parent.right
+            height: lastIndex ? remainingHeight() : nominalHeight()
+            Layout.fillHeight: true
+            Layout.alignment: Qt.AlignRight | Qt.AlignBottom
+
+            property var lastIndex: (index + 1) == numTicks
+
+            Text {
+                id: label
+                //anchors.fill:parent // this text object is not anchored
+                //Layout.fillHeight: true
```

Comment:
This code is for debug or can be deleted ?

Reply:
It is a reminder for anyone who adjusts it in the future. It is a little bit nonconventional how I got it to look right I think.

Reply:
I can try to rework it

Reply:
There are some bugs with what I expect to be the reasonable thing to do (bind text anchors to parent, use horizontal and verticalAlignment). It hangs the program when trying to use these. I left the comments just to explain the situation.